Sources/Core/AWSClientRuntime/Config/AwsDefaultClientConfiguration.swift

```diff
@@ -0,0 +1,36 @@
+//
+// Copyright Amazon.com Inc. or its affiliates.
+// All Rights Reserved.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+public protocol AwsDefaultClientConfiguration {
+    /// The credentials provider to be used for AWS credentials.
+    ///
+    /// If no credentials provider is supplied, the SDK will look for credentials in the environment, then in the `~/.aws/credentials` file.
+    var credentialsProvider: CredentialsProviding { get set }
```

Comment:
Isn't credentials provider a non-AWS Smithy construct, and if so, shouldn't this go in the Smithy `DefaultClientConfiguration`?

Reply:
The `CredentialsProviding` protocol (and all the credentials related classes) currently resides in aws-sdk-swift AWSClientRuntime. 
We can potentially move it into smithy-swift's ClientRuntime and then add it to `DefaultClientConfiguration` in a seperate PR